src/pydvl/influence/frameworks/torch_differentiable.py

```diff
@@ -63,18 +65,19 @@ def num_params(self) -> int:
 
     def split_grad(
         self,
-        x: Union["NDArray", "torch.Tensor"],
-        y: Union["NDArray", "torch.Tensor"],
+        x: torch.Tensor,
+        y: torch.Tensor,
         progress: bool = False,
-    ) -> "NDArray":
+    ) -> NDArray:
         """
-        Calculates gradient of model parameters wrt each x[i] and y[i] and then
+        Calculates gradient of model parameters wrt each $x[i]$ and $y[i]$ and then
         returns a array of size [N, P] with N number of points (length of x and y) and P
         number of parameters of the model.
-        :param x: A np.ndarray [NxD] representing the features x_i.
-        :param y: A np.ndarray [NxK] representing the predicted target values y_i.
+        :param x: A np.ndarray [NxD] representing the features $x_i$.
```

Comment:
A docstring type is redundant because it is already available in the signature, and in this case, it contradicts it